hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -3906,6 +3908,21 @@ public void access(final Path f, final FsAction mode)
   @Retries.RetryTranslated
   public FileStatus getFileStatus(final Path f) throws IOException {
     Path path = qualify(f);
+    if (isTrackMagicCommitsInMemoryEnabled(getConf()) && isMagicCommitPath(path)) {
+      // Some downstream apps might call getFileStatus for a magic path to get the file size.
+      // when commit data is stored in memory construct the dummy S3AFileStatus with correct
+      // file size fetched from the memory.
+      if (InMemoryMagicCommitTracker.getTaskAttemptIdToBytesWritten().containsKey(path)) {
+        long len = InMemoryMagicCommitTracker.getTaskAttemptIdToBytesWritten().get(path);
+        return new S3AFileStatus(len,
```

Comment:
how about we add a special etag here, like "pending", declared in CommitterConstants. That way toString() on the status hints that it is pending so there's more diagnostics in list operations etc. And it could be used in tests.

Reply:
ack